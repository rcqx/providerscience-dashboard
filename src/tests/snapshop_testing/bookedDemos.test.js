import renderer from 'react-test-renderer';
import BookedDemos from '../../components/bookedDemos/bookedDemos';

// Snapshot test for BookedDemos component
it('renders correctly', () => {
  const mockDemos = [
    {
      id: 18565984,
      name: 'Erica Hubbard',
      email: 'erica@providerscience.com',
      total_value: 14351505632,
      photo_url: null,
      goals: {
        generated_demo: [
          {
            month: '2023-03-01T00:00:00.000Z',
            metric: 'GeneratedDemo',
            goal: '10',
            is_circular: true,
            _id: '645d53461fbe1dad12e87dfb',
          },
          {
            month: '2023-05-01T00:00:00.000Z',
            metric: 'GeneratedDemo',
            goal: '15',
            is_circular: true,
            _id: '6463c25132dd3214e58bcfa2',
          },
        ],
      },
      total_deals: 773,
      leads_to_demos: 0,
    },
    {
      id: 18277410,
      name: 'Cole Elledge',
      email: 'cole@providerscience.com',
      total_value: 4642462140,
      photo_url: null,
      goals: {
        generated_demo: [
          {
            month: '2023-03-01T00:00:00.000Z',
            metric: 'GeneratedDemo',
            goal: '10',
            is_circular: true,
            _id: '645d53461fbe1dad12e87dfb',
          },
          {
            month: '2023-05-01T00:00:00.000Z',
            metric: 'GeneratedDemo',
            goal: '15',
            is_circular: true,
            _id: '6463c25132dd3214e58bcfa2',
          },
        ],
      },
      total_deals: 254,
      leads_to_demos: 0,
    },
  ];

  // BookedDemos renders correctly
  const tree = renderer.create(
    <BookedDemos
      allDemos={mockDemos}
      request={true}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();

  // BookedDemos renders correctly with no demos
  const tree2 = renderer.create(
    <BookedDemos
      allDemos={[]}
      request={true}
    />)
    .toJSON();
  expect(tree2).toMatchSnapshot();
});
