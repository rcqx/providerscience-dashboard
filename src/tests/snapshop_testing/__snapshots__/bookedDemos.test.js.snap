// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex-1 xl:mb-0 xl:h-[85%]"
>
  <h1
    className="text-start text-[#4CBEEB] text-xl lg:text-2xl font-bold mb-2 mt-2 tracking-tight"
  >
    Booked Demos
  </h1>
  <div
    className="border border-slate-200 flex py-1 rounded-t-lg"
  >
    <h1
      className="font-bold text-slate-700 md:w-[15%] w-[25%]"
    >
      Rank
    </h1>
    <h1
      className="font-bold text-slate-700 md:w-[20%] w-[25%]"
    >
      Name
    </h1>
    <h1
      className="font-bold text-slate-700 flex-1 lg:pl-2"
    >
      Goal
    </h1>
    <h1
      className="font-bold flex-1 text-slate-700 lg:text-end lg:pr-6"
    >
      Demos
    </h1>
  </div>
  <div
    className="flex flex-col justify-end w-full h-full border-x rounded-b-lg shadow-md"
  />
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="flex-1 xl:mb-0 xl:h-[85%]"
>
  <h1
    className="text-start text-[#4CBEEB] text-xl lg:text-2xl font-bold mb-2 mt-2 tracking-tight"
  >
    Booked Demos
  </h1>
  <div
    className="border border-slate-200 flex py-1 rounded-t-lg"
  >
    <h1
      className="font-bold text-slate-700 md:w-[15%] w-[25%]"
    >
      Rank
    </h1>
    <h1
      className="font-bold text-slate-700 md:w-[20%] w-[25%]"
    >
      Name
    </h1>
    <h1
      className="font-bold text-slate-700 flex-1 lg:pl-2"
    >
      Goal
    </h1>
    <h1
      className="font-bold flex-1 text-slate-700 lg:text-end lg:pr-6"
    >
      Demos
    </h1>
  </div>
  <div
    className="flex flex-col justify-end w-full h-full border-x rounded-b-lg shadow-md"
  >
    <div
      className="flex justify-center items-center h-52 xl:h-full"
    >
      <h1
        className="text-slate-600 text-2xl italic"
      >
        No Booked Demos ðŸ˜¿
      </h1>
    </div>
  </div>
</div>
`;
